### ISSUE TITLE
staticx: Failed to cleanup /tmp/staticx-xxxxxx: Directory not empty

### ISSUE THREAD

**Issue by JonathonReinhart** [JonathonReinhart/staticx#25]
This happens continuously in Travis:

https://travis-ci.org/JonathonReinhart/staticx/jobs/242388537

**Comment by JonathonReinhart** [comment]
It looks like the depth-first traversal simply isn't working correctly under Ubuntu 14.04.5's `musl 0.9.15-1`.

Here's the (good) [build output](https://travis-ci.org/JonathonReinhart/staticx/jobs/242805677) using Glibc:
```
Running staticx executable
Wed Jun 14 12:22:21 UTC 2017
[remove_tree] pathname=/tmp/staticx-CP3fQb
  [remove_tree_fn] fpath=/tmp/staticx-CP3fQb/.staticx.prog typeflag=FTW_F
  [remove_tree_fn] fpath=/tmp/staticx-CP3fQb/ld-2.19.so typeflag=FTW_F
  [remove_tree_fn] fpath=/tmp/staticx-CP3fQb/ld-linux-x86-64.so.2 typeflag=FTW_SL
  [remove_tree_fn] fpath=/tmp/staticx-CP3fQb/libc.so.6 typeflag=FTW_SL
  [remove_tree_fn] fpath=/tmp/staticx-CP3fQb/.staticx.interp typeflag=FTW_SL
  [remove_tree_fn] fpath=/tmp/staticx-CP3fQb/libc-2.19.so typeflag=FTW_F
  [remove_tree_fn] fpath=/tmp/staticx-CP3fQb/archive.tar typeflag=FTW_F
  [remove_tree_fn] fpath=/tmp/staticx-CP3fQb typeflag=FTW_DP
```

And here's the (bad) [build output](https://travis-ci.org/JonathonReinhart/staticx/jobs/242805676) using musl-libc:
```
Running staticx executable under centos:5
Wed Jun 14 12:22:12 UTC 2017
[remove_tree] pathname=/tmp/staticx-nDEflG
  [remove_tree_fn] fpath=/tmp/staticx-nDEflG typeflag=FTW_DP
ERROR: [remove_tree_fn] failed to remove /tmp/staticx-nDEflG: Directory not empty
```

**Comment by JonathonReinhart** [comment]
Ubuntu 14.04 is using a fairly old version of musl (0.9.15). We're suffering from [this bug](https://git.musl-libc.org/cgit/musl/tree/WHATSNEW?id=84eff797#n1173), fixed in version 1.0.0:

> - nftw FTW_MOUNT flag prevented walking any directories at all

Indeed, removing `FTW_MOUNT` in my test program made it work.
